homeassistant/components/lock/zwave.py

```diff
@@ -221,36 +220,30 @@ def __init__(self, value):
 
     def update_properties(self):
         """Callback on data changes for node values."""
-        self._state = self._value.data
+        self._state = self.values.primary.data
         _LOGGER.debug('Lock state set from Bool value and'
                       ' is %s', self._state)
-        notification_data = self.get_value(class_id=zwave.const
-                                           .COMMAND_CLASS_ALARM,
-                                           label=['Access Control'],
-                                           member='data')
-        if notification_data:
+        if self.values.access_control:
+            notification_data = self.values.access_control.data
             self._notification = LOCK_NOTIFICATION.get(str(notification_data))
-        if self._v2btze:
-            advanced_config = self.get_value(class_id=zwave.const
-                                             .COMMAND_CLASS_CONFIGURATION,
-                                             index=12,
-                                             data=CONFIG_ADVANCED,
-                                             member='data')
-            if advanced_config:
-                self._state = LOCK_STATUS.get(str(notification_data))
-                _LOGGER.debug('Lock state set from Access Control '
-                              'value and is %s, get=%s',
-                              str(notification_data),
-                              self.state)
-
-        alarm_type = self.get_value(class_id=zwave.const
-                                    .COMMAND_CLASS_ALARM,
-                                    label=['Alarm Type'], member='data')
+
+            if self._v2btze:
+                if self.values.v2btze_advanced and \
+                        self.values.v2btze_advanced.data == CONFIG_ADVANCED:
+                    self._state = LOCK_STATUS.get(str(notification_data))
+                    _LOGGER.debug('Lock state set from Access Control '
+                                  'value and is %s, get=%s',
+                                  str(notification_data),
+                                  self.state)
+
+        if not self.values.alarm_type:
+            return
+
+        alarm_type = self.values.alarm_type.data
         _LOGGER.debug('Lock alarm_type is %s', str(alarm_type))
-        alarm_level = self.get_value(class_id=zwave.const
-                                     .COMMAND_CLASS_ALARM,
-                                     label=['Alarm Level'], member='data')
+        alarm_level = self.values.alarm_level and self.values.alarm_level.data
```

Comment:
If the first expression evaluates to True, it will return the second expression, although I agree it should be rewritten to be less confusing.